javascript/javascript1/week4/VoiceAssistant.js

```diff
@@ -0,0 +1,180 @@
+const commandHistory = []
+let curUser = ""
+const todos = []
+
+function getReply(command)
+{
+// save the name and respond with a message
+    //Checking whether the user already indtroduced 
+    let found = false;
+    for(item in commandHistory) 
+    {
+        if(commandHistory[item] === command.toLowerCase()) 
+        {
+            found = true;
+            break;
+        }
+    }
+    //If user already exists print a message else save the user name and set current user
+    if(command.toLowerCase().startsWith("hello my name is"))
+     {
+   
+        curUser = command.split(" ").pop()
+        if(curUser == "")
+        {
+            console.log(`No name found. Enter a valid name`)
+        }
+       else if(found)
+        {
+            console.log(`Hello ${curUser}, we know you already`)
+        } 
+        else 
+        {
+            commandHistory.push(command.toLowerCase()) ;
+            console.log(`nice to meet you ${curUser}`) ;
+        }
+    } 
+// If asking for name, return current user
+    else if(command.toLowerCase().startsWith("what is my name")) 
```

Comment:
Since your using `command.toLowerCase` a lot of places, I'd assign it in the top:
`const lowerCaseCommand = command.toLowerCase;`